operator-k8s/src/operator.rs

```diff
@@ -98,20 +93,7 @@ impl Operator {
                         .await?;
                     return Ok(());
                 }
-
-                if !self.config.create_crd
-                    && versions.iter().any(|ver| {
-                        matches!(
-                            compare_versions(ver.name.as_str(), current_version.as_ref()),
-                            Ok(Ordering::Greater),
-                        )
-                    })
-                {
-                    panic!(
-                        "The current XlineCluster CRD version {current_version} is not compatible with higher version on k8s. Please use the latest deployment operator or set --create_crd to true."
-                    );
-                }
-
+                assert!(self.config.create_crd || !versions.iter().any(|ver| ver > &current_version), "The current XlineCluster CRD version {current_version} is not compatible with higher version on k8s. Please use the latest xline-operator or set --create_crd to true.");
```

Comment:
The comparison of this line and line 85 can be simplified to the same comparison

Reply:
There will be a special case of the equals sign (=) here.
```
!any(ver > current_ver)   <=>    all(current_ver >= ver)
where ver in versions
```
This is somewhat different from line 85.
This might seem a bit puzzling. When "assert failed" occurs, then `!any(ver > current_ver) = false`, it follows that `any(ver > current_ver) = true`. This indicates the existence of a version higher than the current one, rather than equal.